src/utility/HCISharedMemTransport.cpp

```diff
@@ -0,0 +1,748 @@
+  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+#if defined(STM32WBxx)
+
+#include "HCISharedMemTransport.h"
+#include "STM32Cube_FW/hw.h"
+
+/* Private variables ---------------------------------------------------------*/
+PLACE_IN_SECTION("MB_MEM1") ALIGN(4) static TL_CmdPacket_t BleCmdBuffer;
+
+PLACE_IN_SECTION("MB_MEM2") ALIGN(4) static uint8_t EvtPool[POOL_SIZE];
+PLACE_IN_SECTION("MB_MEM2") ALIGN(4) static TL_CmdPacket_t SystemCmdBuffer;
+PLACE_IN_SECTION("MB_MEM2") ALIGN(4) static uint8_t
+SystemSpareEvtBuffer[sizeof(TL_PacketHeader_t) + TL_EVT_HDR_SIZE + 255];
+PLACE_IN_SECTION("MB_MEM2") ALIGN(4) static uint8_t
+BleSpareEvtBuffer[sizeof(TL_PacketHeader_t) + TL_EVT_HDR_SIZE + 255];
+PLACE_IN_SECTION("MB_MEM2") ALIGN(4) static uint8_t
+HciAclDataBuffer[sizeof(TL_PacketHeader_t) + 5 + 251];
+
+
+/* global var used as semaphore to control incoming events */
+volatile bool sys_event; /* true : M0 core is already up and running */
+volatile bool sys_resp;
+volatile bool acl_data_on; /* true : sending ACL data in progress, false : send is possible */
+volatile bool data_overflow;
+
+/* buffer to store the received packets */
+volatile uint8_t _rxbuff[BLE_MODULE_SHARED_MEM_BUFFER_SIZE];
+volatile uint16_t _read_index; /* fifo position when reading */
+volatile uint16_t _write_index; /* fifo position when receiving */
+
+/* var of different device steps during init and receiving  */
+volatile bool phase_bd_addr;
+volatile bool phase_tx_power;
+volatile bool phase_reset;
+volatile bool phase_running;
+
+/** Bluetooth Device Address */
+static uint8_t bd_addr_udn[CONFIG_DATA_PUBADDR_LEN];
+
+/* Private functions ---------------------------------------------------------*/
+/**
+ * TL Mailbox synchronisation means
+ */
+
+/*  returns true if sys_event was received, false otherwise */
+static bool sysevt_wait(void)
+{
+  /* sys_event remains false until event is received */
+  for (unsigned long start = millis(); (millis() - start) < BLE_IPCC_TIMEOUT;) {
+    /* Wait for 10sec max - if not return an error */
+    if (sys_event) {
+      break;
+    }
+  }
+
+  if (!sys_event) {
+#if defined(PRINT_IPCC_INFO)
+    printf("ERROR: sys_evt timeout\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+    /*  no event received, timeout occurs */
+    return false;
+  }
+  /*  release immediately, now that M0 runs */
+  return true;
+}
+
+/*  WEAK callbacks from the BLE TL driver - will be called under Interrupt */
+static void sysevt_received(void *pdata)
+{
+  UNUSED(pdata);
+  /* For now only READY event is received, so we know this is it */
+  sys_event = true;
+  /* But later on ... we'll have to parse the answer */
+}
+
+/*  returns true if sysevt was already received, which means M0 core is
+ *  already up and running */
+static bool sysevt_check(void)
+{
+  /*  Check if system is UP and running already */
+  for (unsigned long start = millis(); (millis() - start) < 10;) {
```

Comment:
I guess BLE_IPCC_TIMEOUT can be used here too

Reply:
Here 10 ms is waiting while `BLE_IPCC_TIMEOUT` is 10000.

Reply:
my mistake, consider resolved